mlx/backend/metal/device.cpp

```diff
@@ -19,16 +19,14 @@ namespace mlx::core::metal {
 
 namespace {
 
-// Catch things related to the main-thread static variables
-static std::shared_ptr<void> global_memory_pool = new_scoped_memory_pool();
```

Comment:
This is the initial cause of trouble -- this was improperly nested when this code was dynamically loaded, e.g. in an xctest situation, but potentially many others.

Although this did provide a top level autoreleasepool on the main thread, it simply collected the objects until the termination of the process.